articles/how-to-use-array-operators-in-js/index.md

```diff
@@ -0,0 +1,507 @@
+
+```javascript
+let fruits = ["mangoes","lemons","banana","pineapples"];
+let fruit = fruits.find(fruit => fruit.length === 6);
+console.log(fruit);
+```
+Output.
+
+```bash
+lemons
+```
+
+Since `lemons` is the first fruit to have six letters, the `find()` method returns it.
+
+- **findIndex()**: It returns the index of the first element to meet a particular condition. If the element we are searching for has more than one occurrence, it returns the first index. Considering the previous example, if we want to find the index of the first fruit to have six letters, we use `findIndex()`.
+
+```javascript
+let fruits = ["mangoes","lemons","banana","pineapples"];
+let index = fruits.findIndex(fruit => fruit.length === 6);
+console.log(index);
+```
+Output
+
+```bash
+1
+```
+Since lemons is the first fruit to meet the condition, its index will be returned.
+
+- **forEach()**: It used to execute a function or a callback for each element of an array. It does not change the array meaning that it always returns undefined. In our previous example, we may intend to `console.log` all the fruits and hence we can use `forEach()` and execute it as a callback.
+
+```javascript   
+let fruits = ["mangoes","lemons","banana"];
+fruits.forEach(fruit => console.log(fruit));
+```
+Output
+
+```bash
+mangoes
+lemons
+banana
+```
+- **from()**: It is used to come up with a new array from an iterable object such as an array or a string. For example if we have an array of strings and from there we want to come up with the same fruits but now capitalized, we can use the `from()` method. The method is usually executed on the `Array` instance.
+
+```javascript
+let fruits = ["mangoes","lemons","banana"];
+let new_fruits = Array.from(fruits,fruit => fruit.toUpperCase());
+console.log(new_fruits);
+```
+Output
+
+```bash
+['MANGOES','LEMONS','BANANA']
+```
+
+The `from()` method is executed on the Array instance and receives the iterable object which in this case is `fruits` and a map function. The map function capitlaizes each of the elements in the array.
+
+- **includes()**: It is used to check if a particular array contains the specified element. For example if we want to check if the fruits array contains lemons, we can simply use the `includes()` method. The method returns a `Boolean`.
+
+```javascript
+let fruits = ["mangoes","lemons","banana"];
+let result = fruits.includes("avocado");
+console.log(result);
+```
+
+Output
+
+```bash
+false
+```
+Since we do not have avocado as an element in the fruits array, the result shall hold `false`.
+
+- **indexOf()**: It is used in searching for an element in an array and then returning its index. For example, if we want to search for bananas in our fruits array and get the index we use the `indexOf()` method. If else we do not find the element in the array it shall return `-1`.
```

Comment:
Does this method search for the element or the index of the element. You could phrase it as `It is used in searching for the index of the element passed in...`

Reply:
solved.